addons/netfox/rollback/rollback-synchronizer.gd

```diff
@@ -316,94 +319,95 @@ func _submit_input(input: Dictionary, tick: int):
-			#_logger.warning("non-linear states for brawler 1. Received tick %s while latest tick is %s" % [received_tick, _latest_state_tick])
-
-	# It is guaranteed that the first states are full, until the client sends acks to the server
-	# which enables the server to start sending diffs
-	if (not _has_received_full_state && NetworkTime._initial_sync_done):
-		_has_received_full_state = true
-		_receive_full_state_ack.rpc_id(1, received_tick)
-
-	if (received_state.size() > 0):
-		set_received_state(received_state, received_tick, multiplayer.get_remote_sender_id())
-	elif (NetworkRollback.enable_state_diffs):
-		if (_states.has(received_tick - 1) == false):
-			_logger.warning("non-linear diff states. Received tick %s while latest tick is %s" % [received_tick, _latest_state_tick])
-		if (_states.is_empty()):
-			_logger.error("Received a state without any properties from %s, but missing previous states!" % multiplayer.get_remote_sender_id())
-			breakpoint
-			return
+	var sender_id = multiplayer.get_remote_sender_id()
+	var sanitized = {}
+	for property_name in received_state:
+		var pe = _property_cache.get_entry(property_name)
+		var value = received_state[property_name]
+		var state_owner_id = pe.node.get_multiplayer_authority()
 		
-		for picked_tick in range(_latest_state_tick, received_tick):
-			_states[picked_tick + 1] = _states[picked_tick]
-		_latest_state_tick = received_tick
-
-	else:
-		_logger.error("Received a state without any properties from %s, but diff states which should make it possible, are disabled!" % multiplayer.get_remote_sender_id())
-		breakpoint
-
-func set_received_state(received_state: Dictionary, received_tick: int, sender_id: int):
-	var sanitized: Dictionary = {}
-	var missing_property: bool = false
-	for property_entry in _record_state_props:
-		var state_owner_id: int = property_entry.node.get_multiplayer_authority()
 		if state_owner_id != sender_id:
 			_logger.warning("Received state for node owned by %s from %s, sender has no authority!" \
 				% [state_owner_id, sender_id])
 			continue
-
-		if (received_state.has(property_entry.to_string()) == false):
-			missing_property = true
-			continue
-
-		var value = received_state[property_entry.to_string()]
-		sanitized[property_entry.to_string()] = value
-
-	## Detect if new property is added on runtime and sent
-	var new_property: bool = false
-	for property in received_state:
-		for property_entry in _record_state_props:
-			if (property_entry.to_string() == property):
-				continue
-		new_property = true
-		break
+		
+		sanitized[property_name] = value
 	
 	# Duplicates the previous state(s), including the current one
 	# Also fixes packet loss (e.g. a state missing)
-	if (_latest_state_tick != -1 && NetworkRollback.enable_state_diffs):
+	if (NetworkRollback.enable_state_diffs && _latest_state_tick != -1):
 		for picked_tick in range(_latest_state_tick, received_tick):
 			_states[picked_tick + 1] = _states[picked_tick].duplicate(true)
```

Comment:
In a future issue, I will investigate the default behaviour for not duplicating states. I think that having gaps in the states (via packet losses) will lead to more jitter. I got to confirm it by reading `TickInterpolator`. Until then, the if condition of `NetworkRollback.enable_state_diffs` ensures that even with this PR, there is absolutely no change in behaviour when state diffs are disabled.

After all, the RPC `_submit_full_state` is basically copy-pasted from the original `_submit_state`. Any deviations made from it have an `if (NetworkRollback.enable_state_diffs)` at it, because even with state diffs, full states are still used.